finder-app/writer.c

```diff
@@ -0,0 +1,52 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
+#include <libgen.h>
+#include <syslog.h>
+
+int main(int argc, char* argv[]) {
+    openlog(NULL, 0, LOG_USER);
+
+    //Check for correct number of arguments
+    if (argc != 3) {
+        syslog(LOG_ERR, "Incorrect number arguments: 2 needed\n");
+        return 1; //TODO: look at return codes, put the correct one
+    }
+    
+    int file_fd;
+    int bytes_written;
+
+    char* filepath = strdup(argv[1]);
```

Comment:
@vido2373 you shouldn't need to make your own strdup version here, you could just reference argv[1].